core/opendaq/opendaq/include/opendaq/instance_builder.h

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright 2022-2023 Blueberry d.o.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include <coretypes/stringobject.h>
+#include <opendaq/instance.h>
+
+BEGIN_NAMESPACE_OPENDAQ
+
+/*!
+ * @ingroup opendaq_devices
+ * @addtogroup opendaq_instance InstanceBuilder
+ * @{
+ */
+
+/*!
+ * @brief Builder component of Instance objects. Contains setter methods to configure the Instance parameters, such as Context (Logger, Scheduler, ModuleManager) and RootDevice.
+ * Contains a  `build` method that builds the Instance object.
+ *
+ * The InstanceBuilder provides a fluent interface for setting various configuration options for an Instance
+ * object before its creation. It allows customization of the logger, module manager, scheduler and root device. 
+ * Once configured, calling the `build` method returns a fully initialized Instance object with the specified settings.
+ * 
+ * @subsection Configuration Methods:
+ * The InstanceBuilder provides the following configuration methods:
+ *
+ * - **Logger:** The custom Logger for the Instance. This logger will be used for logging messages related to
+ *   the Instance and its components. When configured, the GlobalLogLevel, ComponentLogLevel and LoggerSink will be ignored, as they are in use only with the default Instance logger
+ *
+ * - **GlobalLogLevel:** The default Logger global log level for the Instance. All log messages with a severity
```

Comment:
I'd lean towards writing the names of properties/builder fields with spaces in-between and only the 1st word capitalized. Eg. **Global log level**. This is how the Property Doxygen documentation was structured.